src/main/scala/com/microsoft/hyperspace/telemetry/HyperspaceEvent.scala

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (2020) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.telemetry
+
+import org.apache.commons.lang.StringUtils
+
+import com.microsoft.hyperspace.index.IndexLogEntry
+
+/**
+ * Trait for all Hyperspace events.
+ */
+trait HyperspaceEvent
+
+/**
+ * General index CRUD event.
+ *
+ * @param sparkUser User of spark application.
+ * @param appId Spark application id.
+ * @param appName Spark application name.
+ * @param index Related indexes.
+ * @param message Message about event.
+ */
+class HyperspaceIndexCRUDEvent(
+    sparkUser: String,
```

Comment:
looks like we can capture `sparkUser`, `appId` and `appName` as `AppInfo`?

Reply:
and it can be used in `protected def event` as well

Reply:
thanks @imback82 , done